src/org/javarosa/core/model/actions/SetValueAction.java

```diff
@@ -96,7 +107,7 @@ public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOExcep
 
     public void writeExternal(DataOutputStream out) throws IOException {
         ExtUtil.write(out, target);
-
+        ExtUtil.write(out, trigger);
```

Comment:
Do you need a corresponding `ExtUtil.read` in `readExternal`?

A broader question: What do we do with already serialized FormDefs when we change the format of the serialized data?

Reply:
Good catch. Perhaps this is a reason to split out `SetNestedValueAction`?

With this structure, there is some chance that someone would build a form with the feature, load it in a version of Collect that doesn't support it, upgrade Collect and then be confused that the feature "doesn't work" because the form was serialized without it. In fact, I think that happened to me the first time I tried this branch. Am I understanding the implications correctly?